src/nvim/lua/vim.lua

```diff
@@ -167,16 +167,15 @@ end
 ---
 --- Example: To remove ANSI color codes when pasting:
 --- <pre>
---- vim.paste = (function()
```

Comment:
It may be more idiomatic Lua to do

```lua
do
  local overriden = vim.paste
  vim.paste = function(lines, phase)
    ...
  end
end
```

This is more of a `lua-style` question, though.

Reply:
Would you like to discuss that over here? #11312

Reply:
@norcalli the "js style" in this PR saves a line. But the `do-end` thing is good to mention somewhere, and makes sense for cases where a function isn't being defined, IMO